From 2ea2e4e502e5840d52e76461e071882cc65f1766 Mon Sep 17 00:00:00 2001
From: Roy Li <rongqing.li@windriver.com>
Date: Thu, 15 Nov 2012 02:27:54 +0000
Subject: [PATCH] bind: make "/etc/init.d/bind stop" work

Upstream-Status: Inappropriate [configuration]

Add some configurations, make rndc command be able to controls
the named daemon.

Signed-off-by: Roy Li <rongqing.li@windriver.com>
---
 conf/named.conf | 5 +++++
 conf/rndc.conf  | 5 +++++
 2 files changed, 10 insertions(+)
 create mode 100644 conf/rndc.conf

diff --git a/conf/named.conf b/conf/named.conf
index 95829cf..021dbca 100644
--- a/conf/named.conf
+++ b/conf/named.conf
@@ -47,3 +47,8 @@ zone "255.in-addr.arpa" {
 // root-delegation-only exclude { "DE"; "MUSEUM"; };
 
 include "/etc/bind/named.conf.local";
+include "/etc/bind/rndc.key" ;
+controls {
+	inet 127.0.0.1 allow { localhost; }
+	keys { rndc-key; };
+};
diff --git a/conf/rndc.conf b/conf/rndc.conf
new file mode 100644
index 0000000..4b43a3d
--- /dev/null
+++ b/conf/rndc.conf
@@ -0,0 +1,5 @@
+include "/etc/bind/rndc.key";
+options {
+	default-server  localhost;
+	default-key     rndc-key;
+};
